pubnub-gson/pubnub-gson-impl/build.gradle.kts

```diff
@@ -29,7 +29,7 @@ dependencies {
     testImplementation(libs.junit.vintage.engine)
     testImplementation(libs.awaitility)
     testImplementation(libs.mockk)
-
+    testImplementation("com.google.guava:guava:33.3.0-jre")
     "integrationTestImplementation"(libs.owner)
 }
 
```

Comment:
Checkstyle configuration(see line 36) is defined in 3 places. Would it be possible to extract it to one place e.g.

in /kotlin/pubnub-gson/build.gradle.kts add: 

```
allprojects {
    plugins.withId("checkstyle") {
        checkstyle {
            toolVersion = "8.14"
            configFile = rootProject.file("config/checkstyle/checkstyle.xml")
            sourceSets = listOf(project.sourceSets.getByName("main"))
        }
    }

    tasks.withType<Checkstyle>().configureEach {
        exclude("**/vendor/**", "**/*Test*")

        reports {
            xml.required = true
            html.required = true
        }
    }
}
```
and delete:
```
checkstyle {
    toolVersion = "8.14"
    configFile = rootProject.file("config/checkstyle/checkstyle.xml")
    sourceSets = listOf(project.sourceSets.getByName("main"))
}

tasks.withType<Checkstyle>().configureEach {
    exclude("**/vendor/**", "**/*Test*")

    reports {
        xml.required = true
        html.required = true
    }
}
```
from:
kotlin/pubnub-gson/pubnub-gson-api/build.gradle.kts
/kotlin/pubnub-gson/pubnub-gson-impl/build.gradle.kts

?

Reply:
this isn't done yet, I'll see if I can move it to a plugin instead. 

PS we should never use `allprojects` / `subprojects` - apply plugins to relevant projects instead

Reply:
done